Marlin/Marlin_main.cpp

```diff
@@ -1589,6 +1657,12 @@ static void setup_for_endstop_move() {
         current_position[X_AXIS], current_position[Y_AXIS], zPosition,
         current_position[E_AXIS]
       );
+      #ifdef REPROBE
+        if (zPosition == MIN_PROBE_PT && !digitalRead(Z_MIN_PIN)^Z_MIN_ENDSTOP_INVERTING) {
```

Comment:
@thinkyhead we do know where the probe is before/on the first probe because we know the machine has already found its origin. Unless I'm mistaken, please elaborate.